src/aq/aq_fields_mod.F90

```diff
@@ -1979,4 +1950,25 @@ subroutine aq_field_from_fieldset(self, vars, fset)
    !
 end subroutine aq_field_from_fieldset
 
+function aq_find_variable_index(self, var_name) result(ib_pos)
+   class(aq_fields), intent(in) :: self
+   character(len=*), intent(in) :: var_name
+   integer(atlas_kind_idx) :: ib_pos
+   !
+   integer(atlas_kind_idx) :: ib_var
+   character(len=aq_strlen) :: message
+   !
+   ib_pos = 0
+   do ib_var = 1, self%n_vars
+      if (self%var_name(ib_var) == var_name) ib_pos = ib_var
```

Comment:
Sounds like GNU 11 has a bad implementation of FINDLOC.  
In your coding you could break the loop (exit in fortran syntax) as soon as the variable is found.